docs/root/overview.rst

```diff
@@ -0,0 +1,97 @@
+
+### BenchmarkClient
+
+As of today, there’s a single implementation called** BenchmarkClientImpl,** which wraps Envoy’s **Upstream** concept and (slightly) customized H1/H2 **Pool** concepts. For executing requests, the pool will be requested to create a **StreamEncoder**, and Nighthawk will pass its own **StreamDecoderImpl** into that as an argument. The integration surface between **BenchmarkClient **is defined via **BenchmarkClient::tryStartRequest()** and a callback specification which will be fired upon completion of a successfully started request.
+
+For H3, it is anticipated that it will fit into this model, but if all else fails, it will be entirely possible to wire in a new type of **BenchmarkClient.**
+
+### RequestSource
+
+**RequestSource **is an abstraction that allows us to implement different ways for **BenchmarkClient** to get information on what the request that it is about to fire off should look like. Today, two implementations exist: a static one, which will repeat the same request over and over, as well as one that pulls dynamic request data from a grpc service. The latter can, for example, be used to implement log-replay.
+
+### StreamDecoder
+
+StreamDecoder is a Nighthawk-specific implementation of an Envoy concept. It is responsible for coordinating lifetime events of a request to upper abstraction layers (**BenchmarkClient**, **Sequencer) **as well as latency recording.
+
+### OutputCollector
+
+OutputCollector is a container that facilitates building up the native output format of Nighthawk (proto3, nighthawk::client::Output). It is the basis for all output formats offered by Nighthawk, including CLI human output.
+
+### OutputFormatter
+
+OutputFormatter is responsible for transformations of nighthawk::client::Output to requested formats (e.g. human, json, fortio, etc)
+
+### Statistic
+
+Nighthawk’s statistic is responsible for administrating latencies. The most notable implemenation that exists today wraps HdrHistogram, but Nighthawk also has a couple of other implementations which mostly exist to ensure that floating point math is correct in tests. For various reasons, HdrHistogram might get replaced by [libcirclhist](https://github.com/envoyproxy/nighthawk/issues/115) in the near future.
+
+### H1Pool & H2Pool
+
+Nighthawk derives its own version of these from the vanilla Envoy ones. It does that to implement things like pro-active connection prefetching and H2 multi-connection support, as well as offer more connection management strategies. 
+
+## Executables
+
+### nighthawk_client
+
+This is the CLI interface
```

Comment:
Can we add something like: ... it is used to control the nighthawk_test_server (or whichever name we choose for it, e.g. Nighthawk Traffic API). If we do change the name, we should apply that throughout this document.

Reply:
Added more text; for the rename I refer to my earlier comment on the subject.